source/client-side-encryption/client-side-encryption.rst

```diff
@@ -1258,24 +1284,39 @@ One of the strings:
 - "AEAD_AES_256_CBC_HMAC_SHA_512-Random"
 - "Indexed"
 - "Unindexed"
+- "RangePreview"
 
-The result of explicit encryption with the "Indexed" algorithm must be processed by the server to insert or query. Drivers MUST document the following behavior:
+The result of explicit encryption with the "Indexed" or "RangePreview" algorithm must be processed by the server to insert or query. Drivers MUST document the following behavior:
 
-   To insert or query with an "Indexed" encrypted payload, use a ``MongoClient`` configured with ``AutoEncryptionOpts``.
+   To insert or query with an "Indexed" or "RangePreview" encrypted payload, use a ``MongoClient`` configured with ``AutoEncryptionOpts``.
    ``AutoEncryptionOpts.bypassQueryAnalysis`` may be true. ``AutoEncryptionOpts.bypassAutoEncryption`` must be false.
 
+NOTE: The Range algorithm is experimental only. It is not intended for public use. It is subject to breaking changes.
+
 contentionFactor
 ^^^^^^^^^^^^^^^^
-contentionFactor only applies when algorithm is "Indexed".
+contentionFactor only applies when algorithm is "Indexed" or "RangePreview".
 It is an error to set contentionFactor when algorithm is not "Indexed".
 
+NOTE: The Range algorithm is experimental only. It is not intended for public use. It is subject to breaking changes.
+
 queryType
 ^^^^^^^^^
 One of the strings:
 - "equality"
+- "rangePreview"
+
+queryType only applies when algorithm is "Indexed" or "RangePreview".
+It is an error to set queryType when algorithm is not "Indexed" or "RangePreview".
 
-queryType only applies when algorithm is "Indexed".
-It is an error to set queryType when algorithm is not "Indexed".
+NOTE: The Range algorithm is experimental only. It is not intended for public use. It is subject to breaking changes.
+
+rangeOpts
+^^^^^^^^^
+rangeOpts only applies when algorithm is "rangePreview".
```

Comment:
what if queryType is `rangePreview`? Are rangeOpts allowed?

Reply:
>  what if queryType is rangePreview? 

queryType="rangePreview" requires algorithm="rangePreview". This is [validated in libmongocrypt](https://github.com/mongodb/libmongocrypt/blob/7f6c763f39a52bf46ce2b2d7c476528e16cce843/src/mongocrypt-ctx-encrypt.c#L2475-L2478).

> Are rangeOpts allowed?

rangeOpts are required when queryType="rangePreview".